pennylane_lightning/src/gates/cpu_kernels/avx_common/ApplyCNOT.hpp

```diff
@@ -60,22 +58,24 @@ template <typename PrecisionT, size_t packed_size> struct ApplyCNOT {
                                       size_t num_qubits,
                                       [[maybe_unused]] bool inverse) {
         constexpr static auto perm =
-            permutationInternalInternal<control, target>();
+            applyInternalInternalPermuation<control, target>();
 
         for (size_t n = 0; n < exp2(num_qubits); n += packed_size / 2) {
             const auto v = PrecisionAVXConcept::load(arr + n);
             PrecisionAVXConcept::store(arr + n, Permutation::permute<perm>(v));
         }
     }
 
-    template <size_t control> static constexpr auto maskInternalExternal() {
-        std::array<bool, packed_size> mask = {
-            false,
-        };
+    template <size_t control>
+    static constexpr auto applyInternalExternalMask() {
+        std::array<bool, packed_size> mask{};
```

Comment:
This is not recommended, but I wonder if `vector<bool>`'s specialization would give any improvement here (as in, storing single bits per bool rather than bytes). The downsides obviously outweigh the benefit (access and breaking certain standard functionalities), but if we there is benefit in reducing the size of the array to 1/8 its original size, this may work.

That being said, the same would hold true for a `std::bitset`

Reply:
This function is only called in the compile-time, so I do not worry about memory usage. In fact, it should be `consteval`, but most of compilers do not support `consteval` yet. I will just add it to `TODO` here.